projects/ngrx-auto-entity/src/lib/factories/util.ts

```diff
@@ -0,0 +1,32 @@
+import { ActionCreator, Creator } from '@ngrx/store';
+import { EntityAction } from '../..';
+import { TNew } from '../actions/model-constructor';
+
+export interface CorrelatedProps {
+  correlationId?: string;
+}
+
+export interface StandardProps extends CorrelatedProps {
+  criteria?: any;
+}
+
+export const NAE_TYPE_ACTION_CACHE = '__nae_type_action_cache';
+export const cacheOnType = <TModel, TAction extends EntityAction<TModel>, T extends string>(
```

Comment:
This function caches created action factories on each entity model's class (the type, not the instances of the type.) In part, this was to try and avoid crating some kind of internal tracking structure, which would have bloated this library more than absolutely necessary.

Beyond that, however, this potentially allows the end developer to DYNAMICALLY access the action factories that may exist for any given entity.

This opens the door for looping through the EntityActionTypes enum and looking for an action factory for any given entity type. I think any way we can increase the ability of the end developer to implement dynamic functionality, the better. I don't know if, when, or how it might be used...but, some times that's the best thing about writing an open source library: To see what people can do with it! :P